sdk/monitor/Azure.Monitor.OpenTelemetry/src/DefaultAzureMonitorOpenTelemetryOptions.cs

```diff
@@ -9,6 +9,7 @@ namespace Azure.Monitor.OpenTelemetry
     internal class DefaultAzureMonitorOpenTelemetryOptions : IConfigureOptions<AzureMonitorOpenTelemetryOptions>
     {
         private const string AzureMonitorOpenTelemetrySectionFromConfig = "AzureMonitorOpenTelemetry";
+        private const string ConnectionStringEnvironmentVariable = "APPLICATIONINSIGHTS_CONNECTION_STRING";
```

Comment:
If this is to take highest preference, then do we need to get it from IConfiguration? Can we just read the env-var and overwrite anything provided via IConfiguration? Otherwise, how do we ensure it takes precedence over some other source, like appsettings.json file

Reply:
> Can we just read the env-var and overwrite anything provided via IConfiguration? 

We are exactly doing the same with the proposed change.

Reply:
_configuration?[ConnectionStringEnvironmentVariable]

^ This can read from anything bound to IConfiguration, not just env variable..

Reply:
IConfiguration reads environment variables by default. There are 2 ways we could read APPLICATIONINSIGHTS_CONNECTION_STRING
1. Using IConfiguraion
2. Environment.GetEnvironmentVariable()

I used IConfiguration as we are in that context.

Reply:
Got it, will change to read from `Environment.GetEnvironmentVariable`

Reply:
> IConfiguration reads environment variables by default

This is not a guarantee. Its done by the default webhost builder, but we can't assume so.

Reply:
![image](https://user-images.githubusercontent.com/5232798/222339770-2afe6869-382e-4f6b-a9f9-23729351a523.png)
My new assistant is doing good job today ;)

Reply:
It's a cool assistant, comes handy nowadays :blush: